### PR TITLE
Create xboxcontroller class

### PR DESCRIPTION
**Pull request by Twinters007**

### CODE REVIEW

File: wpilib/wpilib/xboxcontroller.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    and the mapping of ports to hardware buttons depends on the code in the Driver Station.
+     """
+
+    def __init__(self, port):
+        """Construct an instance of an XBoxController. The XBoxController index is the USB port on the Driver Station.
+
+        :param port: The port on the Driver Station that the joystick is plugged into
+        """
+
+        super(port)
```
Comment (virtuald) [inline wpilib/wpilib/xboxcontroller.py]: `super().__init__(port)`

File: wpilib/wpilib/xboxcontroller.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    def getPOVCount(self):
+        return self.ds.getPOVCount(self.port)
+
+    def getType(self):
+        return self.ds.getJoystickType(self.port)
+
+    def getName(self):
+        return self.ds.getJoystickName(self.port)
+
+    def setOutput(self, *args):
```
Comment (virtuald) [inline wpilib/wpilib/xboxcontroller.py]: What's with this? There's only a single overload on the java version.

File: wpilib/wpilib/xboxcontroller.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        if value > 1:
+            value = 1
+
+        if type_ == RumbleType.kLeftRumble:
+            self.leftRumble = value * 65535
+        else:
+            self.rightRumble = value * 65535
+
+        hal.setJoystickOutputs(
+            bytes(
```
Comment (virtuald) [inline wpilib/wpilib/xboxcontroller.py]: hm?

Comment (Twinters007) [inline wpilib/wpilib/xboxcontroller.py]: autoPep strikes again

**Comment by virtuald** [comment]
You did something weird... why are my joystick et al changes showing up as new content here?

**Comment by Twinters007** [comment]
Oh. I pulled so I could get you stick changes. Will deleting the commits in rebase fix?

**Comment by Twinters007** [comment]
Yes it did

**Comment by virtuald** [comment]
Thanks!

**Comment by auscompgeek** [comment]
> Oh. I pulled so I could get you stick changes.

FYI, the correct thing to do in this scenario is usually to just do a rebase (git has `git pull --rebase` for this as well).

**Pull request merged**